### PR TITLE
[RFC] Added some basic signals docs

### PR DESCRIPTION
**Pull request by JoshBarr**
Starting to address #1764, but needs expanding on. Let me know if I'm getting the gist of this right –  there's probably lots of other obvious uses for signals that I'm not covering yet.

### CODE REVIEW

File: docs/reference/signals.rst [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+--------------
+
+This signal is emitted from a ``PageRevision`` when a revision is set to `published`.
+
+:sender: The page ``class``
+:instance: The specific ``Page`` instance.
+:revision: The ``PageRevision`` that was published
+:kwargs: Any other arguments passed to ``page_published.send()``.
+
+To listen to a signal, implment ``page_published.connect(receiver, sender, **kwargs)``. Here's a simple
```
Comment (tomdyson) [inline docs/reference/signals.rst]: implment->implement

**Comment by tomdyson** [comment]
This is a great start, @JoshBarr, thanks a lot!

File: docs/reference/signals.rst [hunk 2]
```diff
@@ -0,0 +1,88 @@
+.. _signals:
+
+Signals
+=======
+
+Wagtail's `PageRevision <./pages/model_reference#pagerevision>`__  and
```
Comment (kaedroho) [inline docs/reference/signals.rst]: if you link to the classes like this ``:class:`~wagtail.wagtailcore.models.PageRevision` ``. It'll generate the link to the reference section automatically.

**Comment by gasman** [comment]
Looks good to me! Any reason not to merge this as-is?

**Pull request merged**